client/common/cmdline.c

```diff
@@ -1962,6 +1953,51 @@ int freerdp_client_settings_parse_command_line_arguments(rdpSettings* settings,
 		CommandLineSwitchEnd(arg)
 	}
 	while ((arg = CommandLineFindNextArgumentA(arg)) != NULL);
+	
+       
+       // Check if Username is set and Domain is missing.
+       // If this is the case we should split Username and Domain at the @ sign (e.g. for Kerberos)
+       // Else we should let the username string as it is (e.g. for UPN Suffix)
+       if (username && !settings->Domain)
+       {
+	  freerdp_parse_username(username, &user, &domain);
+         
+	  settings->Username = _strdup(user);
+	  settings->Domain = _strdup(domain);
+       }
+       else
+	  settings->Username = _strdup(username);
+       
+	if (user != NULL)
+	{
+		free(user);
+		user = NULL;
+	}
+	if (domain != NULL)
+	{
+		free(domain);
+		domain = NULL;
+	}
+       
+       // Same as above for Gateway Username and Domain.
+       if (gw_username && !settings->GatewayDomain)
+       {
+	  freerdp_parse_username(gw_username, &user, &domain);
+         
+	  settings->GatewayUsername = _strdup(user);
+	  settings->GatewayDomain = _strdup(domain);
+       }
+       else
+	  settings->GatewayUsername = _strdup(gw_username);
+       
+       if (user != NULL)
+	 free(user);
+       if (domain != NULL)
+	 free(domain);
+       if (username != NULL)
+	 free(username);
+       if (gw_username != NULL)
+	 free(gw_username);
```

Comment:
You don't have to check pointers passed to free() for NULL. Please see man free ;)